### PR TITLE
feat(suite): configure coinjoin strategies

### LINKED ISSUE #6522: Custom max mining fee
Solve the communication of `max mining fee` selected in custom setup of CoinJoin between Suite and Trezor device.

### LINKED ISSUE #6625: Display CoinJoin-related values received from coordinator
Display values from `coinjoin.clients` added in #6563.

### PR DESCRIPTION
**Pull request by komret**
Display real values in CoinJoin Strategy.

## Description
This PR is aimed at displaying values received from coordinator (max mining fee) or config (estimated time, max rounds, skip rounds) or both (coordinator fee rate - see discussion in #6563) and feeding them to `TrezorConnect.authorizeCoinjoin`. Some things need to be defined/confirmed before send the PR for review:

1) The Service fee displayed is a simple calculation of `notAnonymized` x `coordinatorFeeRate`. This might be incorrect, coins with anonymity > 1 will not incur any coordinator fee IIRC. Are there any instructions for rounding the fee amount to whole satoshis or should standard rounding be applied?
2) The time estimates for `recommended`, `fast` and `custom` strategy need to be defined. For `custom` strategy, there needs to be some sort of calculation or the value should be omitted.
3) Max rounds for each strategy need to be defined, the current values are probably not final. It is not customizable in `custom` strategy.
4) The range in `custom` strategy only allows values between 1 and 100, while the current `recommended` value is 129.
5) Max mining fee is duplicated in `custom` strategy - green value above range + the grey box.

## Related Issue

Resolve #6625

## Screenshots (if appropriate):
Screenshot 2022-10-21 at 23 28 14

### CODE REVIEW

File: packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx]: get coordinator fee rate from status (state.wallet.coinjoin.clients[network].coordinatorFeeRate)

File: packages/suite/src/utils/wallet/coinjoinUtils.ts
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/coinjoinUtils.ts]: i think i was better before (tranformed data in reducer), you dont need to perform calculations on each render of `CoinjoinDefaultStrategy` component (kvB2vB inside) or any other component which will potentially require those data 

btw this was [your argument ](https://github.com/trezor/trezor-suite/pull/6563#issuecomment-1279027774) and the reason why i've done it :) did you change your mind?

calculation (multiplication) will be done only when calling `actions.startCoinjoinSession` function, you are already doing it when custom feeRate is used

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

File: packages/suite/src/views/wallet/anonymize/components/CoinjoinSessionDetail.tsx
Comment (tomasklim) [inline packages/suite/src/views/wallet/anonymize/components/CoinjoinSessionDetail.tsx]:
```suggestion
    skipRounds?: [number, number];
```
Usable without `null`?

File: packages/suite/src/views/wallet/anonymize/components/CoinjoinDefaultStrategy.tsx
Comment (tomasklim) [inline packages/suite/src/views/wallet/anonymize/components/CoinjoinDefaultStrategy.tsx]: magic function? 🪄  

At least a comment or a better name would be nice

File: packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx
Comment (komret) [reply packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx]: What does this value actually represent and what unit is it? I will add some comment explaining it. If `coordinatorFeeRate` is 0.003, how do I calculate `maxCoordinatorFeeRate`?

Comment (komret) [reply packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx]: Please check d1a44e04ca7838099ddb814645430151d8c745ab

File: packages/suite/src/utils/wallet/coinjoinUtils.ts
Comment (komret) [reply packages/suite/src/utils/wallet/coinjoinUtils.ts]: Yeas, I am torn. Changing it back. I did not like how it gets transformed there and back again, but it seems like a lesser evil. d1a44e04ca7838099ddb814645430151d8c745ab

File: packages/suite/src/views/wallet/anonymize/components/CoinjoinDefaultStrategy.tsx
Comment (komret) [reply packages/suite/src/views/wallet/anonymize/components/CoinjoinDefaultStrategy.tsx]: Moved and comment added in d1a44e04ca7838099ddb814645430151d8c745ab

File: packages/suite/src/views/wallet/anonymize/components/CoinjoinSessionDetail.tsx
Comment (komret) [reply packages/suite/src/views/wallet/anonymize/components/CoinjoinSessionDetail.tsx]: I want it to be nullable explicitly, but the `?` could be removed: 87459d45ae1968470b0095a185dbd52dacd14df7

File: packages/suite/src/utils/wallet/coinjoinUtils.ts
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/coinjoinUtils.ts:87]: maybe lets do the same with the fee rate, lets multiplyply by 100 here and use it in action below

File: packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx]: if you multiply by 100 in the `transformCoinjoinStatus` the you don't have to multiply it here and you can use "percentage format" anywhere else in components

it also would be nice to use some utilit we already have like `amountToSatoshi` but it returs a string :/

File: packages/suite/src/utils/wallet/coinjoinUtils.ts
Comment (komret) [reply packages/suite/src/utils/wallet/coinjoinUtils.ts:87]: The raw value (0.003) is used in `CoinjoinSetupStrategies` to calculate the fee amount so I thought it better to keep it as is.

File: packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx
Comment (komret) [reply packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx]: Yes, that is why I did not use `amountToSatoshi`...

**Comment by hynek-jina** [comment]
@komret Thanks for noting the issues. Let's handle them in next phase. We can keep it as it is for `22.11`

**Comment by komret** [comment]
260de5c4d73f2d7b9afe688ae477f3418b0b47ed introduces some simple calculation of `estimatedTime` and `maxRounds`. It does not take `maxMiningFee` into account, though.

File: packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/anonymize/components/CoinjoinSetupStrategies.tsx:89]: just note for followup. lets create a selector in `@wallet-reducers/coinjoinReducer` which returns all data combined:
- selectedAccount
- coinjoin account related to selected account
- coinjoin session of coijoin account
- Balance Breakdown
- coinjoin client

**Comment by komret** [comment]
466d891c3a397cdea9d3b33209bb615a385d68ad explains `coordinatorMaxFeeRate` and does not use `network.decimals` for its calculation.

It also uses `coordinatorFeeRate` instead of the config value in `CoinjoinAccountEmpty.tsx`. Once we remove the onboarding modals, the fee rate can be removed from config.

@szymonlesisz pls approve, this should be all for this PR.

**Comment by komret** [comment]
@szymonlesisz 296b5a141b414ce6e94239beb4ca4ba18529d782 introduces `getMaxRounds` and `getEstimatedTimePerRound` utilities

**Review by szymonlesisz — APPROVED** [review]
great, i guess we dont have to wait for the decision of discussion regarding `coordinatorFeeRate` in [firmware]( https://github.com/trezor/trezor-firmware/pull/2577#discussion_r1006632951), that branch isnt merged yet and will not work with current setup (2-master) anyway.

we will change that calculation once it will be merged

**Pull request merged**
